### PR TITLE
Updated `glam` to `0.21`.

### LINKED ISSUE #4: Update `glam` to 2.1
Currently, `glam` is at `0.2.1`. 
I tried updating `encase`, and there are no compile errors when upgrading directly, except that `"std"` has to be added to the "features".

### PR DESCRIPTION
**Pull request by CGMossa**
Fixes #4.

### CODE REVIEW

**Comment by devil-ira** [comment]
This will bump the MSRV to `1.58.1`.

File: Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "encase"
 version = "0.2.1"
 edition = "2021"
-rust-version = "1.57"
+rust-version = "1.58.1"
```
Comment (teoxoy) [inline Cargo.toml:5]:
```suggestion
rust-version = "1.58"
```

File: Cargo.toml [hunk 2]
```diff
@@ -39,7 +39,7 @@ const_panic = { version = "0.2", default-features = false }
 
 mint = { version = "0.5.9", default-features = false, optional = true }
 cgmath = { version = "0.18", default-features = false, optional = true }
-glam = { version = "0.20", features = [ "num-traits" ], default-features = false, optional = true }
+glam = { version = "0.21", features = [ "std", "num-traits" ], default-features = false, optional = true }
```
Comment (teoxoy) [inline Cargo.toml:42]:
```suggestion
glam = { version = "0.21", features = [ "std" ], default-features = false, optional = true }
```

**Comment by CGMossa** [comment]
Sure. `num-traits` feature was already there. But maybe that is no longer needed.

**Pull request merged**